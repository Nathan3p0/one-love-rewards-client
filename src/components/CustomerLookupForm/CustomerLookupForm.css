.dashboard__user-lookup--form {
    color: #FDFFFC;
}