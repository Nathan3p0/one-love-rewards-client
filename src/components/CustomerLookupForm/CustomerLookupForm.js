import React from 'react'

const CustomerLookupForm = (props) => {
    return (
        <form onSubmit={props.handlePhoneSubmit} className="dashboard__user-lookup--form">
            <h2>Search for Customer</h2>
            <div role='alert'>
                {props.error && <p className='red'>{props.error.routine}</p>}
            </div>
            <div>
                <label htmlFor="phone">Enter Customer Phone Number:</label>
                <br />
                <input type="tel" id="phone" name="phone_number" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                    placeholder="Phone Number" onChange={props.handleInputChange} value={props.phone} required />
            </div>
            <div>
                <button className="cust-btn" type="submit">Search</button>
            </div>
        </form>
    )
}

export default CustomerLookupForm