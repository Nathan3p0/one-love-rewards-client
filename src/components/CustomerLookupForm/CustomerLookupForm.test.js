import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CustomerLookupForm from './CustomerLookupForm'

describe('Customer lookup form Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<CustomerLookupForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<CustomerLookupForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<CustomerLookupForm
            handlePhoneSubmit={() => { }}
            handleInputChange={() => { }}
            phone={'734-673-5101'}
            error={{ routine: 'error' }}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})