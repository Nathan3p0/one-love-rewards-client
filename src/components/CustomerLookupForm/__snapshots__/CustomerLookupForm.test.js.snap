// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Item Component Rendering Tests renders to UI as expected 1`] = `
<form
  className="dashboard__user-lookup--form"
  onSubmit={[Function]}
>
  <h2>
    Search for Customer
  </h2>
  <div
    role="alert"
  >
    <p
      className="red"
    >
      error
    </p>
  </div>
  <div>
    <label
      htmlFor="phone"
    >
      Enter Customer Phone Number:
    </label>
    <br />
    <input
      id="phone"
      name="phone_number"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="Phone Number"
      required={true}
      type="text"
      value="734-673-5101"
    />
  </div>
  <div>
    <button
      className="cust-btn"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<form
  className="dashboard__user-lookup--form"
  onSubmit={[Function]}
>
  <h2>
    Search for Customer
  </h2>
  <div
    role="alert"
  >
    <p
      className="red"
    >
      error
    </p>
  </div>
  <div>
    <label
      htmlFor="phone"
    >
      Enter Customer Phone Number:
    </label>
    <br />
    <input
      id="phone"
      name="phone_number"
      onChange={[Function]}
      pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
      placeholder="Phone Number"
      required={true}
      type="text"
      value="734-673-5101"
    />
  </div>
  <div>
    <button
      className="cust-btn"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
